### ISSUE TITLE
Setting plot range for fields causes error

### ISSUE THREAD

**Issue by n01r** [openPMD/openPMD-viewer#160]
While using the openPMD-viewer for a quick look at a high-density plasma setup I wanted to reduce the plot range of my field record `e_density`. Since the densities are in SI units they amount up to the order of 1e29 particles/m^{-3}. Overriding the default plot range now leads to an error.

This behavior starts if one puts in an exponent larger than 18 into the form. (See left side of the following screenshot and the error below).

![2017_05_16_openpmd_plot_range_error](https://cloud.githubusercontent.com/assets/5416860/26116799/b0de4b16-3a64-11e7-9162-8e0f514c08eb.png)

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/opmd_viewer/openpmd_timeseries/interactive.py in refresh_field(change, force)
     89                     slicing=slicing_button.value, theta=theta_button.value,
     90                     slicing_dir=slicing_dir_button.value,
---> 91                     vmin=vmin, vmax=vmax, cmap=fld_color_button.value)
     92 
     93         def refresh_ptcl(change=None, force=False):

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/opmd_viewer/openpmd_timeseries/main.py in get_field(self, field, coord, t, iteration, m, theta, slicing, slicing_dir, output, plot, **kw)
    479                 self.plotter.show_field_2d(F, info, slicing_dir, m,
    480                                     field_label, self.geometry,
--> 481                                     self.current_i, **kw)
    482 
    483         # Return the result

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/opmd_viewer/openpmd_timeseries/plotter.py in show_field_2d(self, F, info, slicing_dir, m, field_label, geometry, current_i, **kw)
    227         plt.imshow(F, extent=1.e6 * info.imshow_extent, origin='lower',
    228                    interpolation='nearest', aspect='auto', **kw)
--> 229         plt.colorbar()

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/pyplot.py in colorbar(mappable, cax, ax, **kw)
   2260 
   2261     ret = gcf().colorbar(mappable, cax = cax, ax=ax, **kw)
-> 2262     return ret
   2263 colorbar.__doc__ = matplotlib.colorbar.colorbar_doc
   2264 

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/figure.py in colorbar(self, mappable, cax, ax, use_gridspec, **kw)
   1600             else:
   1601                 cax, kw = cbar.make_axes(ax, **kw)
-> 1602         cax._hold = True
   1603         cb = cbar.colorbar_factory(cax, mappable, **kw)
   1604 

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/colorbar.py in colorbar_factory(cax, mappable, **kwargs)
   1345         cb = ColorbarPatch(cax, mappable, **kwargs)
   1346     else:
-> 1347         cb = Colorbar(cax, mappable, **kwargs)
   1348 
   1349     cid = mappable.callbacksSM.connect('changed', cb.on_mappable_changed)

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/colorbar.py in __init__(self, ax, mappable, **kw)
    924                 kw['alpha'] = mappable.get_alpha()
    925 
--> 926             ColorbarBase.__init__(self, ax, **kw)
    927 
    928     def on_mappable_changed(self, mappable):

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/colorbar.py in __init__(self, ax, cmap, norm, alpha, values, boundaries, orientation, ticklocation, extend, spacing, ticks, format, drawedges, filled, extendfrac, extendrect, label)
    324         # The rest is in a method so we can recalculate when clim changes.
    325         self.config_axis()
--> 326         self.draw_all()
    327 
    328     def _extend_lower(self):

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/colorbar.py in draw_all(self)
    345         and do all the drawing.
    346         '''
--> 347         self._process_values()
    348         self._find_range()
    349         X, Y = self._mesh()

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/colorbar.py in _process_values(self, b)
    684             self.norm.vmin, self.norm.vmax = mtrans.nonsingular(self.norm.vmin,
    685                                                                 self.norm.vmax,
--> 686                                                                 expander=0.1)
    687 
    688             b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))

/home/.../PARATAXIS_ENV_PY3/lib/python3.4/site-packages/matplotlib/transforms.py in nonsingular(vmin, vmax, expander, tiny, increasing)
   2771     close to zero, it returns -*expander*, *expander*.
   2772     '''
-> 2773     if (not np.isfinite(vmin)) or (not np.isfinite(vmax)):
   2774         return -expander, expander
   2775 

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

I think this is connected to #64. An option that allows for rescaling of the data to a suitable unit system would avoid numbers that are too large.

**Comment by RemiLehe** [comment]
Thanks a lot for reporting this bug.

I agree that in the long term, we should have a better unit system (cf #64). For the short term though, this particular error is easy to fix: I just realized that the slider generates the `vmin` and `vmax` of the plot as `int32` (which is why it overflows for "reasonably small" numbers like 5e19) ! Using a float instead fixes the issue. I'll do a corresponding pull request later today.

**Comment by n01r** [comment]
Cool! Much appreciated, thanks! ✨

**Comment by ax3l** [comment]
fixed in `dev` with #161
